Content.Server/Flash/FlashSystem.cs

```diff
@@ -28,18 +22,40 @@ internal sealed class FlashSystem : SharedFlashSystem
         [Dependency] private readonly IGameTiming _gameTiming = default!;
         [Dependency] private readonly StunSystem _stunSystem = default!;
         [Dependency] private readonly InventorySystem _inventorySystem = default!;
+        [Dependency] private readonly MetaDataSystem _metaSystem = default!;
         [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
 
         public override void Initialize()
         {
             base.Initialize();
-
             SubscribeLocalEvent<FlashComponent, MeleeHitEvent>(OnFlashMeleeHit);
             SubscribeLocalEvent<FlashComponent, MeleeInteractEvent>(OnFlashMeleeInteract);
             SubscribeLocalEvent<FlashComponent, UseInHandEvent>(OnFlashUseInHand);
             SubscribeLocalEvent<FlashComponent, ExaminedEvent>(OnFlashExamined);
+
             SubscribeLocalEvent<InventoryComponent, FlashAttemptEvent>(OnInventoryFlashAttempt);
+
             SubscribeLocalEvent<FlashImmunityComponent, FlashAttemptEvent>(OnFlashImmunityFlashAttempt);
+
+            SubscribeLocalEvent<FlashableComponent, ComponentStartup>(OnFlashableStartup);
+            SubscribeLocalEvent<FlashableComponent, ComponentShutdown>(OnFlashableShutdown);
+            SubscribeLocalEvent<FlashableComponent, MetaFlagRemoveAttemptEvent>(OnMetaFlagRemoval);
+        }
+
+        private void OnMetaFlagRemoval(EntityUid uid, FlashableComponent component, ref MetaFlagRemoveAttemptEvent args)
+        {
+            if (component.LifeStage > ComponentLifeStage.Initialized) return;
+            args.Cancelled = true;
+        }
+
+        private void OnFlashableStartup(EntityUid uid, FlashableComponent component, ComponentStartup args)
+        {
+            _metaSystem.AddFlag(Comp<MetaDataComponent>(uid), MetaDataFlags.EntitySpecific);
```

Comment:
Use the proxy instead
```suggestion
            _metaSystem.AddFlag(MetaData(uid), MetaDataFlags.EntitySpecific);
```